src/nav2_util/src/Costmap.cpp

```diff
@@ -0,0 +1,180 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <vector>
+
+#include "nav2_util/Costmap.hpp"
+#include "tf2/LinearMath/Quaternion.h"
+
+using std::vector;
+
+namespace nav2_util
+{
+
+const Costmap::CostValue Costmap::no_information = 255;
+const Costmap::CostValue Costmap::lethal_obstacle = 254;
+const Costmap::CostValue Costmap::inscribed_inflated_obstacle = 253;
+const Costmap::CostValue Costmap::medium_cost = 128;
+const Costmap::CostValue Costmap::free_space = 0;
+
+// TODO(orduno): Port ROS1 Costmap package
+Costmap::Costmap(rclcpp::Node * node)
+: node_(node)
+{
+  RCLCPP_INFO(node_->get_logger(), "Costmap::Costmap");
+}
+
+nav2_tasks::msg::Costmap
+Costmap::getCostmap(const nav2_tasks::msg::CostmapMetaData & /*specifications*/)
+{
+  // TODO(orduno): build a costmap given the specifications
+
+  RCLCPP_INFO(node_->get_logger(), "Costmap::getCostmap");
+
+  // TODO(orduno): faking out a costmap for now
+
+  nav2_tasks::msg::Costmap costmap;
+  costmap.header.stamp = node_->now();
+  costmap.header.frame_id = "/map";
+
+  // Fill map metadata
+  costmap.metadata.map_load_time = node_->now();
+
+  // Some arbitrary numbers
+  costmap.metadata.resolution = 1;  // m/cell
+  costmap.metadata.size_x = 10;  // cells
+  costmap.metadata.size_y = 10;  // cells
+
+  // The origin of the map [m, m, rad]. This is the real-world pose of the cell (0,0) in the map.
+  // Origin is lower-left pixel?
+  costmap.metadata.origin.position.x = 0.0;  // 2D pose of the lower-left pixel in the map
+  costmap.metadata.origin.position.y = 0.0;
+  costmap.metadata.origin.position.z = 0.0;  // ignored
+
+  // Define map rotation
+  // Provided as yaw with counterclockwise rotation, with yaw = 0 meaning no rotation
+
+  tf2::Quaternion quaternion;
+  quaternion.setRPY(0.0, 0.0, 0.0);  // set roll, pitch, yaw
+  costmap.metadata.origin.orientation.x = quaternion.x();
+  costmap.metadata.origin.orientation.y = quaternion.y();
+  costmap.metadata.origin.orientation.z = quaternion.z();
+  costmap.metadata.origin.orientation.w = quaternion.w();
+
+  costmap.data.resize(costmap.metadata.size_x, costmap.metadata.size_y);
+
+  // Fill with some fake data for testing
+  costmap.data = getTestData(costmap.metadata.size_x, costmap.metadata.size_y);
+
+  return costmap;
+}
+
+vector<uint8_t>
+Costmap::getTestData(const int /*size_x*/, const int /*size_y*/)
+{
+  // TODO(orduno): fixed size for now
+
+  // TODO(orduno): besides hardcoded costmaps, use a mathematical function
+
+  const uint8_t n = no_information;
+  const uint8_t x = lethal_obstacle;
+  const uint8_t i = inscribed_inflated_obstacle;
+  const uint8_t u = medium_cost;
+  const uint8_t o = free_space;
+
+  // TODO(orduno): make these vector of vectors, select maze by index
+  //               within the test program
+
+  vector<uint8_t> costmapFree =
+    // 0 1 2 3 4 5 6 7 8 9
+  {o, o, o, o, o, o, o, o, o, o,    // 0
```

Comment:
Unfortunately, this seems to be the format that cpplint wants.

Reply:
:disappointed: